errbot/templating.py

```diff
@@ -26,24 +27,37 @@ def make_templates_from_plugin_path(plugin_path):
     return make_templates_path(os.sep.join(plugin_path.split(os.sep)[:-1]))
 
 
-def add_plugin_templates_path(path):
+def add_plugin_templates_path(path, identifier=None):
     global env
     tmpl_path = make_templates_from_plugin_path(path)
     if os.path.exists(tmpl_path):
         log.debug("Templates directory found for this plugin [%s]" % tmpl_path)
+        plugin_templates = [tmpl_path, system_templates_path]
+        if identifier is not None:
+            plugin_paths[identifier] = FileSystemLoader(plugin_templates)
         template_path.append(tmpl_path)  # for webhooks
```

Comment:
I think adding to `template_path` might be unnecessary. It keeps everything backwards compatible though.